modules/core/src/main/java/org/apache/ignite/internal/processors/metastorage/persistence/DmsDataWriterWorker.java

```diff
@@ -81,134 +90,125 @@ public DmsDataWriterWorker(
         super(igniteInstanceName, "dms-writer", log);
         this.lock = lock;
         this.errorHnd = errorHnd;
+
+        pauseTask = new FutureTask<>(() -> AWAIT);
+        // Completed future task.
+        pauseTask.run();
+
+        // Put restore task to the queue, so it will be executed on worker start.
+        updateQueue.offer(newDmsTask(this::restore));
     }
 
     /** */
     public void setMetaStorage(ReadWriteMetastorage metastorage) {
         this.metastorage = metastorage;
     }
 
-    /** */
-    public void update(DistributedMetaStorageHistoryItem histItem) {
-        updateQueue.offer(histItem);
+    /**
+     * @return Future which will be completed will all the tasks prior to the pause task completed.
+     */
+    public Future<?> flush() {
+        return pauseTask;
     }
 
-    /** */
-    public void update(DistributedMetaStorageClusterNodeData fullNodeData) {
-        assert fullNodeData.fullData != null;
-        assert fullNodeData.hist != null;
-
-        updateQueue.clear();
+    /**
+     * @param compFut Future which should be completed when worker may proceed with updates.
+     */
+    public void pause(IgniteInternalFuture<?> compFut) {
```

Comment:
may be it would be better to name method ``suspend``

Reply:
Fixed.